integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -1,77 +1,261 @@
+import com.sequenceiq.it.cloudbreak.dto.telemetry.TelemetryTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.log.Log;
 import com.sequenceiq.it.cloudbreak.testcase.e2e.AbstractE2ETest;
+import com.sequenceiq.it.cloudbreak.util.CloudFunctionality;
+import com.sequenceiq.it.cloudbreak.util.FreeIpaInstanceUtil;
+import com.sequenceiq.it.cloudbreak.util.SdxUtil;
 import com.sequenceiq.it.cloudbreak.util.spot.UseSpotInstances;
 import com.sequenceiq.it.cloudbreak.util.ssh.action.SshEnaDriverCheckActions;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+import com.sequenceiq.sdx.api.model.SdxDatabaseAvailabilityType;
+import com.sequenceiq.sdx.api.model.SdxDatabaseRequest;
 
+/**
+ * Google Cloud related features are not completed yet. So this E2E test suite is applicable for AWS and Azure right now.
+ */
 public class DistroXEncryptedVolumeTest extends AbstractE2ETest {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistroXEncryptedVolumeTest.class);
+
+    public String environmentName;
+
     @Inject
     private DistroXTestClient distroXTestClient;
 
     @Inject
-    private TestParameter testParameter;
+    private SshEnaDriverCheckActions sshEnaDriverCheckActions;
 
     @Inject
-    private SshEnaDriverCheckActions sshEnaDriverCheckActions;
+    private CredentialTestClient credentialTestClient;
+
+    @Inject
+    private EnvironmentTestClient environmentTestClient;
+
+    @Inject
+    private AwsCloudProvider awsCloudProvider;
+
+    @Inject
+    private AzureCloudProvider azureCloudProvider;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private FreeIpaInstanceUtil freeIpaInstanceUtil;
+
+    @Inject
+    private SdxUtil sdxUtil;
+
+    @Inject
+    private SdxTestClient sdxTestClient;
 
     @Override
     protected void setupTest(TestContext testContext) {
+        notSupportedCloudPlatform(CloudPlatform.GCP);
         testContext.getCloudProvider().getCloudFunctionality().cloudStorageInitialize();
         createDefaultUser(testContext);
         initializeDefaultBlueprints(testContext);
         createDefaultCredential(testContext);
-        createEnvironmentWithFreeIpaAndDatalake(testContext);
     }
 
-    @Test(dataProvider = TEST_CONTEXT)
+    @Test(dataProvider = TEST_CONTEXT, description = "KNOWN ISSUE: CB-15601 Azure Disk Encryption Set creation throws" +
+            " PSQLException if environment name too long")
     @UseSpotInstances
     @Description(
             given = "there is a running cloudbreak",
-            when = "a valid DistroX create request is sent with encrypted discs",
-            then = "DistroX cluster is created")
+            when = "create an environment then a valid distroX with disk encryption",
+            then = "freeIpa, sdx and distroX volumes should be encrypted with provided keys")
     public void testCreateDistroXWithEncryptedVolume(TestContext testContext) {
```

Comment:
`AzureEnvironmentWithCustomerManagedKeyTests` gets deleted below, which is fine, because its `testWithEncryptionKeyResourceGroup()` seems to be covered here. I wonder though if we shall have another test case as well for the scenario originally tested by `testWithEnvironmentResourceGroup()` (i.e. single-RG, DES and all the clusters located in a common RG)?

I third test case would be an extension of the above single-RG one, such that even the Key Vault was located in the same RG as the env.

Reply:
When I was running these test cases I've realised the `resourceGroupForTest` is empty and does not contain any resource during the entire test execution. So the behaviour on the Azure seems exactly same in case of `testWithEnvironmentResourceGroup` and `testWithEncryptionKeyResourceGroup`. Testing point of view the only difference the `.withResourceGroup(ResourceGroupTest.AZURE_RESOURCE_GROUP_USAGE_SINGLE, resourceGroupForTest)` step. So based on the experienced behaviour on the provider side (Azure Resource Groups) and the 1 step difference on the test case side, I did not take into account the `testWithEnvironmentResourceGroup` test case.

New test case or the possible fox for the single-RG can be part of a follow-up JIRA for the [CB-11741 Nigthly e2e Azure tests should use encryption](https://jira.cloudera.com/browse/CB-11741) with [PR#11898](https://github.com/hortonworks/cloudbreak/pull/11898)